tfx/components/example_gen/utils.py

```diff
@@ -209,84 +213,183 @@ def _glob_to_regex(glob_pattern: Text) -> Text:
   regex_pattern = regex_pattern.replace(')', '\\)')
   return regex_pattern
 
-
-def _retrieve_latest_span(uri: Text,
-                          split: example_gen_pb2.Input.Split) -> Text:
-  """Retrieves the most recently updated span matching a given split pattern."""
+def _glob_and_match_span_version(uri: Text,
+                                 split: example_gen_pb2.Input.Split,
+                                 is_match_version: bool
+                                 ) -> Tuple[Text, Optional[Text]]:
+  """Search for most recently updated properties given by `spec_list`."""
   split_pattern = os.path.join(uri, split.pattern)
-  if split_pattern.count(SPAN_SPEC) != 1:
-    raise ValueError('Only one {SPAN} is allowed in %s' % split_pattern)
 
-  split_glob_pattern = split_pattern.replace(SPAN_SPEC, '*')
+  split_glob_pattern = split_pattern
+  split_regex_pattern = _glob_to_regex(split_pattern)
+
+  split_glob_pattern = split_glob_pattern.replace(SPAN_SPEC, '*')
+  split_regex_pattern = split_regex_pattern.replace(SPAN_SPEC,
+      '(?P<{}>.*)'.format(SPAN_PROPERTY_NAME))
+
+  if is_match_version:
+    split_glob_pattern = split_glob_pattern.replace(VERSION_SPEC, '*')
+    split_regex_pattern = split_regex_pattern.replace(VERSION_SPEC,
+        '(?P<{}>.*)'.format(VERSION_PROPERTY_NAME))
+
   logging.info('Glob pattern for split %s: %s', split.name, split_glob_pattern)
-  split_regex_pattern = _glob_to_regex(split_pattern).replace(SPAN_SPEC, '(.*)')
   logging.info('Regex pattern for split %s: %s', split.name,
                split_regex_pattern)
-  if re.compile(split_regex_pattern).groups != 1:
-    raise ValueError('Regex should have only one group')
+
+  num_groups = re.compile(split_regex_pattern).groups
```

Comment:
do we assume span is always the first group and version is the second? if so we should assert that

Reply:
I spoke with Hui about this, and was told that there aren't any guarantees that Span is first group and Version is second (i.e. a pattern like `ver{VERSION}/span{SPAN}/*`. We don't assume this in our implementation since we're using named groups to retrieve our Span and Version implementation rather than relying on order.

Reply:
my bad, I noticed that but forgot to remove my comment